src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -217,7 +217,7 @@ def dynamic_stub(channel, service, cardinalities, options=None):
   Returns:
     A face.DynamicStub with which RPCs can be invoked.
   """
-    effective_options = StubOptions() if options is None else options
+    effective_options = _EMPTY_STUB_OPTIONS if options is None else options
```

Comment:
Oh well... True believer in static analysis I have always been. Was just too charitable to Python users incorrectly assuming they _must_ have some magic powers, hence their choice of the language. Burned by optimism again! _C'est la vie_... 🏳